### DISCUSSION TITLE
如何在react应用程序中引入Waline

### DISCUSSION THREAD

**Discussion by CLCK0622** [walinejs/waline#1711] [Q&A] [answered]
我最近用nextra生成了一个网站，但是因为nextjs不暴露html代码，需要将Waline用react组件的形式传入。
我不是很懂react，有没有大佬能帮忙看看。
我看了[Waline的文档](https://waline.js.org/cookbook/import/project.html)，也看了[之前的一个讨论中的方法](https://montage.bigdreamer.cc/components/react-waline-client/)。后者在我通过 npm 安装时遇到了问题，而前者对我来说不够详细。
具体而言，我怎么样在网站中引入文档中完成的那个组件？而且它也没有给我输入server URL的地方。

**Comment by lizheming** [comment, marked as answer]
https://github.com/orgs/walinejs/discussions/1045

**Reply by CLCK0622** [reply]
看到了 封装好之后在哪里用呢？
谢谢

**Reply by CLCK0622** [reply]
@lizheming

**Reply by Mister-Hope** [reply]
This is an open source project and we have no SUPPORT here. It's rude to force others teaching you how to do something because of lacking of basics on yourself.

**Reply by CLCK0622** [reply]
I felt especially sorry for this and I succeeded in achieving this with the docs of React this morning. 
Anyway, many thanks for your help in your power.
